### DISCUSSION TITLE
Using LibIIO to read/write registers BUT device does NOT have debug attr: direct_reg_access

### DISCUSSION THREAD

**Discussion by zambezie14** [analogdevicesinc/libiio#938] [Q&A]
I would like to read/write device registers on a sensor. I am able to use iio_device_reg_write() and iio_device_reg_read() if the device has the debug attr: direct_reg_access; however, this device does not have this. What is the recommended best practice for reading and writing registers where this debug attr is not included? Thank you in advance.

**Comment by mhennerich** [comment]
Well, in case the driver doesn't expose this attribute in debugfs, there is not way to access registers.
